Created new navigation menu for user related items

abujeda opened the pull request:
New navigation menu for user related items.
This replaces the `Logged as` and `Log Out` top level navigation items.

Fixes: https://github.com/OSC/ondemand/issues/2486

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1202821104799596/1203781012316634) by [Unito](https://www.unito.io)

adaybujeda commented:
This is how they look in my local environment:

<img width="300" alt="Screenshot 2023-01-19 at 11 19 05" src="https://user-images.githubusercontent.com/17336210/213429406-e51ea1fc-d4fe-46d4-9098-d390d69fdaa4.png">

<img width="300" alt="Screenshot 2023-01-19 at 11 19 16" src="https://user-images.githubusercontent.com/17336210/213429411-183ac940-d19e-445f-ac85-6987692df2fe.png">

johrstrom commented:
I'm a I'd rather not/ 🤷‍♂️ on this. I'm not a UI/UX expert, and I haven't fully internalized the books I've read but there is a cognitive penalty for another menu.  That is, the more menus you have the more the user has to remember (or find in the first place). It seems small, and it kinda is, but even so, I wonder what the value add is in having another menu.

@gerald-byrket wanted this so I'll leave it to him.

gerald-byrket commented:
> I'm a I'd rather not/ 🤷‍♂️ on this. I'm not a UI/UX expert, and I haven't fully internalized the books I've read but there is a cognitive penalty for another menu. That is, the more menus you have the more the user has to remember (or find in the first place). It seems small, and it kinda is, but even so, I wonder what the value add is in having another menu.
> 
> @gerald-byrket wanted this so I'll leave it to him.

Hi @abujeda .  I do want this. However, I thought we were going to hold off on this for further discussion.  Let's discuss in today's meeting please.  We won't be adding another menu as much as we'll be replacing "Logout" with this "User" menu.

gerald-byrket commented:
As per our meeting today. This functionality will be made optional.

abujeda commented:
I have added a profile based feature toggle to enable the new user menu dropdown.
The property is `new_user_menu`. I added a comment that it should be removed for the next release `2.2`

Should be enough time to decide whether to make it the default or remove it.

abujeda commented:
Just a note: as this was mentioned by Gerald in Harvard <-> OSC catchup on 25 Jan, the OnDemand profiles can be configured for one to have the new user menu so that users can play with it and provide feedback.

For automatic profile selection with hostnames, this will require a separate hostname to trigger the new profile and show the new user menu.
Eg:
- https://ondemand.osc.edu => standard interface
- https://beta.ondemand.osc.edu  => new user menu interface

abujeda commented:
Fixed conflicts

gerald-byrket reviewed (commented):
Hey Aday.

This looks & works great.  The only question that I have is should we put "Manage Federated Identy" under the User, instead of the Help?

Thanks,
-gerald

abujeda commented:
> The only question that I have is should we put "Manage Federated Identy" under the User, instead of the Help?

I think so. This is already implemented in this PR.
I believe that `Manage Federated Identity` link is managed by one of the following links (both now under the user menu):
```
 <%= nav_link(t('dashboard.nav_help_change_password'), "key", passwd_url, new_tab: true) %>
 <%= nav_link(t('dashboard.nav_help_two_factor'), "mobile-alt", configure_2fa_url, new_tab: true) %>
```

gerald-byrket reviewed (approved):
Looks good. Thanks!

gerald-byrket commented:
Hey Aday.

Are you going to fix the branch conflicts?

abujeda commented:
Working on fixing the conflicts

abujeda commented:
Conflict fixed, PR ready to go 🚀

gerald-byrket reviewed (changes requested):
Tests aren't completing.

abujeda commented:
As far as I can see the tests completed successfully on the last run, but for some reason the GitHub UI was not displaying as completed:

<img width="1237" alt="Screenshot 2023-03-25 at 10 04 52" src="https://user-images.githubusercontent.com/17336210/227710732-de6f555d-3609-4088-b631-00d17779c0f9.png">

I have rebased from main just to trigger another build.

abujeda commented:
tests completed after rebase - ready to go 🚀

johrstrom commented:
Accidentally closed this in Asana with the unito sync. Now that Gerald's gone, I guess I have to make a decision on this.

It seems late in the release cycle now to add things (even if this is backward compatible) so I'm going to have a think on in and let you know next week may what we'll do with this.

johrstrom commented:
Hey, sorry for delays on delays on delays, but I don't think we're going to move forward with this.
